client/src/components/auth-modal-form.tsx

```diff
@@ -23,6 +23,31 @@ import { SimpleLoader } from "../design-system/components/simple-loader";
 import { useIsWaitingForTooLong } from "../hooks/use-is-waiting-for-too-long";
 import { User } from "@/lib/storage/domain";
 import { getUserService } from "@/domains/user/user-service";
+import { Eye, EyeOff } from "lucide-react";
+
+// Reusable component for a password field with visibility toggle
+const PasswordInput = ({ ...props }: React.ComponentProps<typeof Input>) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  return (
+    <div className="relative">
+      <Input type={showPassword ? "text" : "password"} {...props} />
+      <Button
+        type="button"
+        variant="ghost"
+        size="sm"
```

Comment:
peut-êre que icon est plus adapté

```suggestion
        size="icon"
```